src/main/java/hudson/plugins/ec2/EC2Cloud.java

```diff
@@ -1153,12 +1154,12 @@ public FormValidation doCheckSshKeysCredentialsId(@QueryParameter String value)
          * @throws IOException
          * @throws ServletException
          */
-        protected FormValidation doTestConnection(URL ec2endpoint, boolean useInstanceProfileForCredentials, String credentialsId, String sshKeysCredentialsId, String roleArn, String roleSessionName, String region)
+        protected FormValidation doTestConnection(@AncestorInPath ItemGroup context, URL ec2endpoint, boolean useInstanceProfileForCredentials, String credentialsId, String sshKeysCredentialsId, String roleArn, String roleSessionName, String region)
```

Comment:
Keeping the former method, without the context, to keep the compatibility would be good. Some subclasses may rely on the old signature.

Reply:
Since this is a method expected to be called from jellies, it shouldn't break anything.